index.js

```diff
@@ -144,13 +146,288 @@ function hash(msgVal) {
   return '%' + ssbKeys.hash(encode(msgVal)).replace('.sha256', '.bbmsg-v1')
 }
 
-// FIXME: might split this out and add validateBatch
-function validateSingle(bmsg, previous) {}
+/**
+ * Validate a decoded bendy-butt message value.
+ *
+ * @param {Object} msgVal - an object compatible with ssb/classic `msg.value`
+ * @param {Object} previousMsg - a decoded `msgVal` object
+ * @param {Buffer | string | null} hmacKey - a valid hmac key for signature verification
+ * @returns {Object | string} an `Error` object with descriptive message or success string
+ */
+function validateSingle(msgVal, previousMsg, hmacKey) {
+  const {
+    author,
+    sequence,
+    previous,
+    timestamp,
+    signature,
+    content,
+    contentSignature,
+  } = msgVal
+
+  if (!ref.isFeedType(author))
+    return new Error(
+      `invalid message: author is "${author}", expected a valid feed identifier`
+    )
+
+  if (sequence < 1)
+    return new Error(
+      `invalid message: sequence is "${sequence}", expected a value greater than or equal to 1`
+    )
+
+  const previousErr = validatePrevious(author, sequence, previous, previousMsg)
+  if (previousErr) return previousErr
+
+  if (typeof timestamp !== 'number')
```

Comment:
```suggestion
  if (typeof timestamp !== 'number' || isNaN(timestamp) || !isFinite(timestamp))
```

You're gonna hate this, but `typeof NaN` is `"number"`. :laughing:

Reply:
😬